Add Amos's guide to build a Rust service (#176)
nix-community/awesome-nix: README.md

### Learning

> * [Building a Rust service with Nix](https://fasterthanli.me/series/building-a-rust-service-with-nix) - About creating a Rust application but treats a lot of Nix as well and is *very* thorough.

Reviewer: ```suggestion
* [Building a Rust service with Nix](https://fasterthanli.me/series/building-a-rust-service-with-nix) - An in-depth blog series about creating a Rust application with Nix.
```
